### ISSUE TITLE
new-cap rule flagging React SFC as invalid

### ISSUE THREAD

**Issue by justin808** [eslint/eslint#6876]
**What version of ESLint are you using?**
3.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel

**Please show your full configuration:**
using the latest airbnb react config

**What did you do? Please include the actual source code causing the issue.**
```js
import React, { PropTypes } from 'react';

const href = 'https://github.com/shakacode/react_on_rails/blob/master/README.md#multiple-react-' +
  'components-on-a-page-with-one-store';

const CommentsCount = (props) => (
  <li>
    <a id="js-comment-count" href={href}>
      Comments: {props.commentsCount}
    </a>
  </li>
);

CommentsCount.propTypes = {
  commentsCount: PropTypes.number.isRequired,
};

export default CommentsCount;
```
and reference to that:

```js
import classNames from 'classnames';
import React, { PropTypes } from 'react';

import CommentsCount from './CommentsCount';
import * as paths from '../../constants/paths';

const NavigationBar = (props) => {
  const { commentsCount, pathname } = props;

  return (
    <nav className="navbar navbar-default" role="navigation">
      <div className="container">
        <div className="navbar-header">
          <button
            type="button"
            className="navbar-toggle"
            data-toggle="collapse"
            data-target="#bs-example-navbar-collapse-1"
          >
            <span className="sr-only">Toggle navigation</span>
            <span className="icon-bar" />
            <span className="icon-bar" />
            <span className="icon-bar" />
          </button>
          <a className="navbar-brand" href="http://www.shakacode.com">ShakaCode</a>
        </div>
        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
          <ul className="nav navbar-nav">
            <li className={classNames({ active: (pathname === paths.ROUTER_PATH) })}>
              <a href={paths.ROUTER_PATH}>React Router Demo</a>
            </li>
            <li className={classNames({ active: (pathname === paths.NO_ROUTER_PATH) })}>
              <a href={paths.NO_ROUTER_PATH}>React Demo</a>
            </li>
            <li className={classNames({ active: (pathname === paths.SIMPLE_REACT_PATH) })}>
              <a href={paths.SIMPLE_REACT_PATH}>Simple React</a>
            </li>
            <li className={classNames({ active: (pathname === paths.RAILS_PATH) })}>
              <a href={paths.RAILS_PATH}>Classic Rails</a>
            </li>
            <li>
              <a
                href={
                  'https://github.com/' +
                  'shakacode/react-webpack-rails-tutorial'
               }
              >
                Source on Github
              </a>
            </li>
            <li>
              <a
                href={
                  'http://www.railsonmaui.com/' +
                  'blog/2014/10/03/integrating' +
                  '-webpack-and-the-es6-transpiler' +
                  '-into-an-existing-rails-project/'
               }
              >Tutorials</a>
            </li>
            <li>
              <a
                href={
                  'http://forum.shakacode.com/' +
                  't/fast-rich-client-rails-development' +
                  '-with-webpack-and-the-es6-transpiler/82/22'
               }
              >Forum</a>
            </li>
            {commentsCount && CommentsCount({ commentsCount })}
          </ul>
        </div>
      </div>
    </nav>
  );
};

NavigationBar.propTypes = {
  commentsCount: PropTypes.number,
  pathname: PropTypes.string.isRequired,
};

export default NavigationBar;
```

**What did you expect to happen?**
No error since this is a SFC, required to be capitalized by React.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/Users/justin/shakacode/react-on-rails/react-webpack-rails-tutorial/client/app/bundles/comments/components/NavigationBar/NavigationBar.jsx
  70:31  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
```

**Comment by vitorbal** [comment]
Hmm, I don't think this is really a bug since ESLint doesn't have (and shouldn't have) any knowledge about React.
I wonder if this wouldn't be more suited as a custom rule for [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react).

**Comment by platinumazure** [comment]
The rule is behaving correctly here. If you don't want all capitalized function names to be required to be constructors, use the `capIsNew: false` option. If you want exceptions to the rule, add `"CommentsCount"` and other exceptions to the `capIsNewExceptions` array option.

See the [options section](http://eslint.org/docs/rules/new-cap#options) of the [new-cap documentation](http://eslint.org/docs/rules/new-cap) for more details.

**Comment by ilyavolodin** [comment]
Closing, as the question has been answered. As noted above, the rule is doing the right thing in this case.
